Dockerfile

```diff
@@ -37,15 +37,17 @@ RUN PLACE_COMMIT=$PLACE_COMMIT \
 # Add source last for efficient caching
 COPY src /app/src
 
-# Build App
-RUN mkdir -p /app/bin/drivers
+# Build application
 RUN crystal build --error-trace --release --debug -o bin/core src/app.cr
 
 # These provide certificate chain validation where communicating with external services over TLS
 ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
 
+# Create binary directory
+RUN mkdir -p /app/bin/drivers
+
 # Use an unprivileged user.
-RUN chown appuser:appuser -R /app
+RUN chown appuser -R /app
```

Comment:
I don't think this will work. You can't mutate an area mapped by a volume at a later point in the Dockerfile. Or rather, you can, but you will not see the impact of this as the volume will be overlayed.

Also, heads up general fs changes like this will also lead to larger than necessary images. Layers are copy-on-write so this would lead to two complete copies of `/app` baked into the final image.

Reply:
I think it should work in the case where no volume is mounted and the volume declaration is removed from the dockerfile - as the idea is that only the parts of the volume that should be writeable have the permissions changed.

Then when using named volumes (I assume declared externally) these permissions are maintained in the persisted volume